packages/server/src/core/initTRPC.ts

```diff
@@ -136,7 +143,67 @@ function createTRPCInner<TParams extends PartialRootConfigTypes>() {
         );
       }
     }
+    const router = createRouterFactory<$Config>(config);
+
+    type ClassToProcRouterRecord<TRouter extends object> = Router<
+      RouterDef<
+        $Config,
+        {
+          [TKey in keyof TRouter]: TRouter[TKey] extends
+            | AnyProcedure
+            | AnyRouter
+            ? TRouter[TKey]
+            : never;
```

Comment:
@sachinraja do you know if there's a performant/lazy way of filtering `never` values here?

Reply:
How about this:

```ts
type ClassToProcRouterRecord<TRouter extends object> = Router<
      RouterDef<
        $Config,
        {
          [TKey in keyof TRouter as TRouter[TKey] extends AnyProcedure | AnyRouter ? TKey : never]: TRouter[TKey];
        }
      >
    >;
```